test/mock/createAPIResponse.ts

```diff
@@ -1,21 +1,39 @@
 import { SearchForFacetValues, Response, MultiResponse } from 'algoliasearch';
 
-export const createSingleSearchResponse = (
-  args: Partial<Response> = {}
-): Response => ({
-  hits: [],
-  nbHits: 0,
-  page: 0,
-  nbPages: 0,
-  hitsPerPage: 4,
-  processingTimeMS: 1,
-  query: '',
-  params: '',
-  index: 'index_name',
-  ...args,
-});
+export function createSingleSearchResponse<THit = any>(
+  subset: Partial<Response<THit>> = {}
+): Response<THit> {
+  const {
+    page = 0,
+    hitsPerPage = 20,
+    nbHits = 0,
+    nbPages,
+    processingTimeMS = 0,
+    hits = [],
+    query = '',
+    params = '',
+    exhaustiveNbHits = true,
+    exhaustiveFacetsCount = true,
+    ...rest
+  } = subset;
+  const fallbackNbPages =
+    nbPages !== undefined ? nbPages : Math.ceil(nbHits / hitsPerPage);
```

Comment:
That's kind of meta but it could be worth testing this test util since many of our tests depend on it.

Reply:
agree, but not sure if this is really worth testing, should we test only that line?

Reply:
That's the one where there's more logic so I'd say this is the most "critical". No strong opinion on the rest though.